File: Week3/Code/get_TreeHeight.R
#!usr/bin/env Rscript

# script: get_TreeHeight.R
# Desc: calculates heigfht of tree from angle of elevation and distance
# Author: David Bridgwood (dmb2417@ic.ac.uk)

rm(list = ls())

# required packages

###############################################################################
# get_TreeHeight.R
###############################################################################

# This script takes in a given csv fike from the command line
# calculates heights of trees from the angle
# elevation and the distance from the base using the trigonometric
# formula: height = distance * tan(radians)
#
# ARGUMENTS:
#       degrees     The angle of elevation
#       distance    The distance from the base
#
# OUTPUT:
#       The height of the tree, same units as "distance"


# commandArgs reads in the arguments given in the command line
args <- commandArgs(trailingOnly = TRUE)

Trees <- read.csv(args[1])

# define function
TreeHeight <- function(degrees, distance){
    radians <- degrees * pi / 180
    height <- distance * tan(radians)

    return(height)
}

# add heights as new column
Trees$Tree.m.height <- TreeHeight(Trees$Angle.degrees, Trees$Distance.m)

# write to file
# basename removes the directory, strsplit removes the .csv
outfile <- paste("../Results/", strsplit(basename(args[1]), ".csv"),
                 "_treeheights.csv", sep = "")


write.csv(Trees, outfile, row.names = FALSE)

File: Week3/Code/get_TreeHeight.py
#!/bin/usr/env python3

"""python script to calculates tree heights from the angle of elevation and the
distance fom the base using the trigonometric formula
height = distance * tan(radians)
Author: David Bridgwood"""

__author__ = 'David Bridgwood (dmb2417@ic.ac.uk)'
__version__ = '0.0.1'

# imports
import math
import csv

# function to calculate height from angle and distance

def tree_height(degrees, distance):
    radians = degrees * math.pi /180
    height = distance * math.tan(radians)

    return height

with open("../Data/trees.csv",'r') as trees:

